server/src/session.ts

```diff
@@ -131,6 +131,25 @@ export class Session {
     // project as dirty to force update the graph.
     project.markAsDirty();
     this.info(`Enabling Ivy language service for ${project.projectName}.`);
+
+    // Send diagnostics since we skipped this step when opening the file
+    // (because language service was disabled while waiting for ngcc).
+    // First, make sure the Angular project is complete.
+    this.runGlobalAnalysisForNewlyLoadedProject(project);
+    project.refreshDiagnostics();  // Show initial diagnostics
+  }
+
+  /**
+   * Invoke the compiler for the first time so that external templates get
+   * matched to the project they belong to.
+   */
+  private runGlobalAnalysisForNewlyLoadedProject(project: ts.server.Project) {
+    if (!project.hasRoots()) {
+      return;
+    }
+    const fileName = project.getRootScriptInfos()[0].fileName;
+    // Getting semantic diagnostics will trigger a global analysis.
```

Comment:
Is it that getting semantic diagnostics always triggers global analysis or is this just because we're triggering diagnostics on a root file?

Reply:
Getting semantic diagnostics always triggers global analysis, because a new compiler is created every time. Technically any file will do, not just root files, but the file must be part of the project, so root file is a convenient candidate.